pkg/json/parse.go

```diff
@@ -111,19 +112,49 @@ func (ing *Parser) parseDoc(ctx parserContext, input jsonom.Node) (*ParsedDocNod
 	return ing.parseValue(ctx, input.(*jsonom.Value))
 }
 
+var PolyHintBlock = func(*jsonom.KeyValue) {}
+
 func (ing *Parser) parseObject(ctx parserContext, input *jsonom.Object) (*ParsedDocNode, error) {
 	// An object node
 	if ctx.schemaNode != nil {
 		if !ctx.schemaNode.HasLabel(ls.AttributeTypeObject) {
 			return nil, ls.ErrSchemaValidation{Msg: fmt.Sprintf("An object is expected here but found %s", ctx.schemaNode.GetLabels()), Path: ctx.path.Copy()}
 		}
 	}
+	if ing.discriminator == nil {
+		ing.discriminator = make(map[*lpg.Node][]*lpg.Node)
+	}
+
```

Comment:
Some comments here explaining the use od discriminator, and how cache is used would be good.